File: base/src/main/java/com/zhushenwudi/base/wifi/WifiManager.java
package com.zhushenwudi.base.wifi;

import android.content.Context;

import java.util.List;

public class WifiManager extends BaseWifiManager {

    private WifiManager(Context context) {
        super(context);
    }

    public static IWifiManager create(Context context) {
        return new WifiManager(context);
    }

    @Override
    public boolean isOpened() {
        return manager.isWifiEnabled();
    }

    @Override
    public void openWifi() {
        if (!manager.isWifiEnabled())
            manager.setWifiEnabled(true);
    }

    @Override
    public void closeWifi() {
        if (manager != null && manager.isWifiEnabled())
            manager.setWifiEnabled(false);
    }

    @Override
    public void scanWifi() {
        manager.startScan();
    }

    @Override
    public boolean disConnectWifi() {
        return manager.disconnect();
    }

    @Override
    public boolean connectEncryptWifi(IWifi wifi, String password) {
        if (manager.getConnectionInfo() != null && wifi.SSID().equals(manager.getConnectionInfo().getSSID()))
            return true;
        int networkId = WifiHelper.configOrCreateWifi(manager, wifi, password);
        boolean ret = manager.enableNetwork(networkId, true);
        modifyList(wifi.SSID(), "开始连接...", false);
        return ret;
    }

    @Override
    public boolean connectSavedWifi(IWifi wifi) {
        int networkId = WifiHelper.configOrCreateWifi(manager, wifi, null);
        boolean ret = manager.enableNetwork(networkId, true);
        modifyList(wifi.SSID(), "开始连接...", false);
        return ret;
    }

    @Override
    public boolean connectOpenWifi(IWifi wifi) {
        boolean ret = connectEncryptWifi(wifi, null);
        modifyList(wifi.SSID(), "开始连接...", false);
        return ret;
    }

    @Override
    public boolean removeWifi(IWifi wifi) {
        boolean ret = WifiHelper.deleteWifiConfiguration(manager, wifi);
        updateWifiList();
        return ret;
    }

    @Override
    public List<IWifi> getWifi() {
        return wifis;
    }

}

File: base/src/main/java/com/zhushenwudi/base/wifi/WifiManager.kt
package com.zhushenwudi.base.wifi

import android.content.Context
import com.zhushenwudi.base.wifi.WifiHelper.configOrCreateWifi
import com.zhushenwudi.base.wifi.WifiHelper.deleteWifiConfiguration
import com.zhushenwudi.base.wifi.BaseWifiManager
import dev.utils.LogPrintUtils

class WifiManager private constructor(context: Context) : BaseWifiManager(context) {
    override val isOpened: Boolean
        get() = manager?.isWifiEnabled ?: false
    override val wifi: List<IWifi?>
        get() = wifis

    override fun openWifi() {
        if (manager?.isWifiEnabled == false) {
            manager?.isWifiEnabled = true
        }
    }

    override fun closeWifi() {
        if (manager?.isWifiEnabled == true) {
            manager?.isWifiEnabled = false
        }
    }

    override fun scanWifi() {
        manager?.startScan()
    }

    override fun disConnectWifi(): Boolean {
        return manager?.disconnect() ?: false
    }

    override fun connectEncryptWifi(wifi: IWifi?, password: String?): Boolean {
        manager?.run {
            val ssid = wifi?.SSID() ?: return false
            if (connectionInfo != null && ssid == connectionInfo.ssid) return true
            val networkId = configOrCreateWifi(this, wifi, password)
            val ret = enableNetwork(networkId, true)
            receiverModifyList(ssid, "开始连接...", false)
            return ret
        }

        return false
    }

    override fun connectSavedWifi(wifi: IWifi?): Boolean {
        manager?.run {
            val ssid = wifi?.SSID() ?: return false
            val networkId = configOrCreateWifi(this, wifi, null)
            val ret = enableNetwork(networkId, true)
            receiverModifyList(ssid, "开始连接...", false)
            return ret
        }
        return false
    }

    override fun connectOpenWifi(wifi: IWifi?): Boolean {
        val ssid = wifi?.SSID() ?: return false
        val ret = connectEncryptWifi(wifi, null)
        receiverModifyList(ssid, "开始连接...", false)
        return ret
    }

    override fun removeWifi(wifi: IWifi?): Boolean {
        manager?.run {
            wifi?.let {
                val ret = deleteWifiConfiguration(this, it)
                receiverUpdateWifiList()
                return ret
            }
        }
        return false
    }

    companion object {
        fun create(context: Context): IWifiManager {
            return WifiManager(context).create()
        }
    }
}